src/sentry/api/endpoints/organization_teams.py

```diff
@@ -142,25 +141,44 @@ def post(self, request: Request, organization, **kwargs) -> Response:
 
         :pparam string organization_slug: the slug of the organization the
                                           team should be created for.
-        :param string name: the optional name of the team.
-        :param string slug: the optional slug for this team.  If
-                            not provided it will be auto generated from the
-                            name.
+        :qparam string name: the optional name of the team.
+        :qparam string slug: the optional slug for this team. If not provided it will be auto
+                             generated from the name.
+        :qparam bool setTeamAdmin: If this is true, the user is added to the as a Team Admin
+                                instead of regular member
         :auth: required
         """
         serializer = TeamPostSerializer(data=request.data)
 
-        if serializer.is_valid():
-            result = serializer.validated_data
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+        result = serializer.validated_data
+        set_team_admin = result.get("setTeamAdmin")
+
+        if set_team_admin:
+            if not features.has("organizations:team-roles", organization) or not features.has(
+                "organizations:team-project-creation-all", organization
+            ):
+                return Response(
+                    {"detail": "You do not have permission to join a new team as a team admin"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            if not self.should_add_creator_to_team(request):
+                return Response(
+                    {"detail": "You must be authenticated to join a new team as a Team Admin"},
+                    status=status.HTTP_401_UNAUTHORIZED,
+                )
+
+        # Wrap team creation and member addition in same transaction
+        with transaction.atomic():
```

Comment:
I don't think it's that important to make the entire thing atomic. Basically since this is a new team, this is the concern that we have for race conditions
1. two requests are trying to make the same team. the previous atomic block prevents this
2. two requests are trying to add the same user to the same team. At worst, one of the request will return a 409.

Reply:
My concern was that we create the `Team` but fail to create the `OrganizationMemberTeam` that sets the user as a Team Admin, so we have a rouge team floating around and someone created a project without the proper permissions. Is this a valid concern or can we switch back to the original atomic block without worrying about this?

Reply:
How would a project be able to be created if `OrganizationMemberTeam` row failed to be created? Does the project creation API not check if the caller is a member of the team?

IMO, the use of atomic here is fine, but you may need to reorganize some of the code. I'll point out what in some new comments.